anndata/compat/__init__.py

```diff
@@ -53,6 +53,18 @@ def __repr__():
 
 try:
     from dask.array import Array as DaskArray
+    import collections.abc as cabc
+
+    # If this was on _core/index.py there would be a circular import.
+    # So registering the _subset function with the import here.
+
+    @_subset.register(DaskArray)
+    def _subset_dask(a: DaskArray, subset_idx):
+        if all(isinstance(x, cabc.Iterable) for x in subset_idx):
+            subset_idx = np.ix_(*subset_idx)
+            return a.vindex[subset_idx]
+        return a[subset_idx]
+
```

Comment:
See [dask.array.Array.vindex](https://docs.dask.org/en/stable/generated/dask.array.Array.vindex.html)

I had to register the _subset here. I am not sure if this is the best practice.

Reply:
Could you instead do:

```python
from anndata.compat import DaskArray

...

@_subset.register(DaskArray)
def _subset_dask(a: DaskArray, subset_idx) -> DaskArray:
    ...
```

In `anndata._core.index`?

Reply:
Ah seeing now, what's the circular import?

Reply:
This has been handled in the awkward array PR, I think you can just copy what was done there: https://github.com/scverse/anndata/pull/647/files#diff-14bdd0423d47482c6db457fa70c6118cff28bbc3675a8e309c962b6c209f157b

Reply:
Fixed it on my recent commit. I will now focus on concatenation.